api/envoy/service/discovery/v3/discovery.proto

```diff
@@ -107,6 +128,22 @@ message DiscoveryResponse {
 
   // The control plane instance that sent the response.
   config.core.v3.ControlPlane control_plane = 6;
+
+  // [#not-implemented-hide:]
+  // An optional per-resource status response that can be used when no matching resource(s)
+  // are sent to the client, and will describe the reason. This will be a map
+  // from a resource name to its status. A resource that doesn't appear in the map
+  // implies the resource response is valid. No status map implies that all the
+  // resources in the response are valid.
+  map<string, DiscoveryResponseStatus> resources_status = 7;
```

Comment:
As discussed IRL, we should probably move this to a separate PR for this feature in general. @markdroth would this help get rid of the "15 second rule"?

Reply:
Yeah, I think something like this is a good idea.  But I do think it should probably be a separate PR.

Do we actually want this to be a separate map?  An alternative would be to embed the `DiscoveryResponseStatus` inside of the `Resource` message in the normal `resources` field.  That way, clients just need to look at one field for every resource, and some of those resources might be a special type that indicates that the server didn't have the requested resource.

Also, what is the expected semantic here when a subscribed resource does not exist and multiple responses are sent during the time it does not exist?  For LDS and CDS, is the server obligated to send every non-existent resource in every response, just like it normally does?  For other resource types, can the server send the response status just once, and the client assumes that the status doesn't change until/unless it receives a subsequent response that either populates the resource or changes its response status?  (The semantic here might be a bit more obvious if the `DiscoveryResponseStatus` was inside of the `resources` field instead of being a separate map.)

Reply:
I agree that it makes more sense to add the `DiscoveryResponseStatus` to each specific resource (in `Resource`).
I'm probably missing something here, but IIUC the `resources` field in a SotW response is of type `google.protobuf.Any`, and the response is not required to return the resources in a wrapped `Resource`. If this implies that the `Resource` wrapper is required how do we plan to the gradual introduction of this to existing xDS services?